sdk/couchbase-core/tests/common/test_config.rs

```diff
@@ -24,6 +25,39 @@ lazy_static! {
     pub static ref LOGGER_INITIATED: AtomicBool = AtomicBool::new(false);
 }
 
+// Shared log capture buffer for log message assertions in core tests
+pub static LOG_CAPTURE: LazyLock<Mutex<String>> = LazyLock::new(|| Mutex::new(String::new()));
+static CAPTURE_LOGS: AtomicBool = AtomicBool::new(false);
+
+fn enable_log_capture() {
```

Comment:
Tests within a test file generally run in parallel. If more than 1 test is using log capture then this might do some weird things.